dpnp/dpnp_iface_indexing.py

```diff
@@ -81,7 +81,11 @@ def choose(x1, choices, out=None, mode='raise'):
     --------
     :obj:`take_along_axis` : Preferable if choices is an array.
     """
-    if not use_origin_backend(x1):
+    x1 = dpnp.array(x1)
```

Comment:
We don't need to wrap `x1` to `array`, we expect `x1` is already `array`. Otherwise we should fallback to numpy.